bin/dfx-token-deploy

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+# Original source: https://github.com/dfinity/ckBTC-Minter-Frontend/blob/master/local_deploy.sh
+# Via:             https://github.com/dfinity/nns-dapp/blob/main/scripts/ckbtc/deploy-ckbtc
+set -euo pipefail
+SOURCE_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+PATH="$SOURCE_DIR:$PATH"
+. "$SOURCE_DIR/versions.bash"
+
+print_help() {
+  cat <<-EOF
+
+	Installs ledger and index canisters for a given token.
+
+	EOF
+}
+
+# Source the clap.bash file ---------------------------------------------------
+source "$SOURCE_DIR/clap.bash"
+# Define options
+clap.define short=n long=network desc="The dfx network to use" variable=DFX_NETWORK default="local"
+clap.define short=p long=prefix desc="Prefix for the local canister names" variable=LOCAL_PREFIX default="mytoken_"
+clap.define long=token desc="The token name/symbol" variable=TOKEN default="MyToken"
+clap.define short=y long=yes desc="Deploy even if there are existing canisters." variable=DFX_YES nargs=0
+clap.define short=c long=check desc="Check that the canisters are present and correct." variable=DFX_CHECK nargs=0
+clap.define short=m long=mode desc="Canister install mode." variable=DFX_MODE default="reinstall"
+# Source the output file ----------------------------------------------------------
+source "$(clap.build)"
+
+: Check preconditions. This is always run.
+"${SOURCE_DIR}/dfx-token-import" --check --prefix "$LOCAL_PREFIX" || {
+  echo "ERROR: $TOKEN canister data and files are not all present and correct."
+  echo "       Please run:"
+  echo "       dfx-token-import --prefix '$LOCAL_PREFIX'"
+  echo
+  exit 1
+} >&2
+
+deploy_token() {
+  # We create some canisters ahead of time because we need to know their IDs
+  # before deploying. Remaining canisters will be created when deployed.
+  dfx canister create "${LOCAL_PREFIX}ledger" --network "$DFX_NETWORK" || echo "${LOCAL_PREFIX}ledger canister already exists"
+
+  LEDGERID="$(dfx canister id "${LOCAL_PREFIX}ledger" --network "$DFX_NETWORK")"
+  echo "$LEDGERID"
+
+  echo "Step 1: deploying ledger canister..."
+  DFX_IDENTITY_PRINCIPAL="$(dfx identity get-principal)"
+  dfx deploy "${LOCAL_PREFIX}ledger" --network "$DFX_NETWORK" --argument "(variant {
+  Init = record {
+     token_symbol = \"$TOKEN\";
+     token_name = \"Token $TOKEN\";
+     minting_account = record { owner = principal \"$DFX_IDENTITY_PRINCIPAL\" };
+     transfer_fee = 10;
+     max_memo_length = opt 64;
+     metadata = vec {};
+     initial_balances = vec { record { record { owner = principal \"$DFX_IDENTITY_PRINCIPAL\"; }; 10_000_000; }; };
+     archive_options = record {
+         num_blocks_to_archive = 10_000;
+         trigger_threshold = 20_000;
+         controller_id = principal \"$DFX_IDENTITY_PRINCIPAL\";
```

Comment:
Is this the same account as the ICP faucet? https://github.com/dfinity/nns-dapp/blob/d8568bd0c944a50e731fac29d8f638f4fc330234/frontend/src/lib/api/dev.api.ts#L31

Reply:
No. This is the principal of the identity installing the canisters. So you can mint using a `dfx` command.
My plan was to then mint a lot of tokens into the faucet account (next PR).
It seems slightly more generic this way but if you're worried that the faucet will run dry, we can also make it the minting account.

Reply:
Ah, ok, no, it's ok. Whatever you think it's easier.

We can change later as well. We would only need to hardcode the faucet account principal here, right?

Reply:
Yes, hardcoding the faucet address here would actually work because (and this surprised me) there is no separate method for minting. Minting is done by asking for a transfer from a minting account.